### PR TITLE
bumping version to 2023.06 and masking OpenSSL to older than v3.*

### PR DESCRIPTION
**Pull request by trz42**
This is the first attempt to build a new compatibility layer for EESSI ... ah, an we are curious if the bot can build it or not.

### CODE REVIEW

**Comment by trz42** [comment]
Check if the bot is installed for this repository:

bot: help

File: ansible/playbooks/roles/compatibility_layer/defaults/main.yml [hunk 1]
```diff
@@ -26,6 +26,9 @@ prefix_mask_packages: |
   # stick to GCC 9.x; using a too recent compiler in the compat layer complicates stuff in the software layer,
   # see for example https://github.com/EESSI/software-layer/issues/151
   >=sys-devel/gcc-10
+  # see https://github.com/EESSI/software-layer/issues/257
+  # and https://github.com/EESSI/software-layer/issues/258
+  <dev-libs/openssl-3
```
Comment (boegel) [inline ansible/playbooks/roles/compatibility_layer/defaults/main.yml:31]: This should be `>=` (see the masking for GCC above)

fixed in https://github.com/trz42/compatibility-layer/pull/2

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by trz42** [comment]
Testing updated bot instance:

bot: help
bot: showconfig
bot: build arch:magic inst:cloud

**Comment by boegel** [comment]
was merged via #188

**Pull request merged**